src/esp/assets/GenericInstanceMeshData.cpp

```diff
@@ -135,30 +296,44 @@ Cr::Containers::Optional<InstancePlyData> parsePly(
 }  // namespace
 
 std::vector<std::unique_ptr<GenericInstanceMeshData>>
-GenericInstanceMeshData::fromPLY(Mn::Trade::AbstractImporter& importer,
-                                 const std::string& plyFile,
-                                 const bool splitMesh) {
+GenericInstanceMeshData::fromPLY(
+    Mn::Trade::AbstractImporter& importer,
+    const std::string& plyFile,
+    const bool splitMesh,
+    std::vector<Magnum::Vector3ub>& colorMapToUse,
+    const std::shared_ptr<scene::SemanticScene>& semanticScene) {
   Cr::Containers::Optional<InstancePlyData> parseResult =
-      parsePly(importer, plyFile);
+      parsePly(importer, plyFile, colorMapToUse, semanticScene);
+
   if (!parseResult) {
     return {};
   }
 
   std::vector<GenericInstanceMeshData::uptr> splitMeshData;
-  if (splitMesh && parseResult->objIdsFromPly) {
-    const InstancePlyData& data = *parseResult;
-    std::unordered_map<uint16_t, PerObjectIdMeshBuilder> objectIdToObjectData;
-    for (size_t i = 0; i < data.cpu_ibo.size(); ++i) {
-      const uint32_t globalIndex = data.cpu_ibo[i];
-      const uint16_t objectId = data.objectIds[globalIndex];
-      if (objectIdToObjectData.find(objectId) == objectIdToObjectData.end()) {
+  if (splitMesh &&
+      (parseResult->objIdsFromPly || parseResult->objPartitionsFromSSD)) {
+    std::vector<uint16_t> meshPartitionIds = parseResult->objIdsFromPly
```

Comment:
not supposed to :smile:

Reply:
This is by the way why `Containers::Array` was made non-copyable -- to prevent accidents like this. And writing an explicit `Utility::copy()` is annoying enough to make you think twice if you really need to copy :)